index.bs

```diff
@@ -0,0 +1,210 @@
+
+    1. Let <var>elements</var> be the list of current elements in <var>doc</var>.
+    1. Let <var>unstable elements</var> be an empty list.
+    1. Let <var>total width</var> be the current <a>viewport</a> width.
+    1. Let <var>total height</var> be the current <a>viewport</a> height. 
+    1. For each <var>element</var> of <var>elements</var>:
+        1. Set <a>currentFrameStartingPoint</a> to the <a>starting point</a> of <var>element</var>.
+        1. Set <a>currentVisualRepresentation</a> to the <a>visual representation</a> of <var>element</var>.
+        1. Let <var>unstable</var> be the boolean returned from calling <a>identify an unstable element</a> with <var>element</var>’s <a>currentFrameStartingPoint</a> and <a>previousFrameStartingPoint</a>. 
+        1. Set <var>shiftFraction</var> of the element to:
+            1. 0 if <a>previousFrameStartingPoint</a> is null.
+            1. The max distance the element has moved in any direction computed as the max(abs(<a>previousFrameStartingPoint</a>'s height - <a>currentFrameStartingPoint</a>'s height) / <var>total height</var> , abs(<a>previousFrameStartingPoint</a>'s width - <a>currentFrameStartingPoint</a>'s width) / <var>total width</var>).
+        1. If <var>unstable</var> is true, add a tuple to <var>unstable elements</var> with:
+            1. <a>currentVisualRepresentation</a> of <var>element</var>
+            1. <a>previousVisualRepresentation</a> of <var>element</var>
+            1. <var>shiftFraction</var> of <var>element</var>:
+        1. Set <a>previousFrameStartingPoint</a> point to <a>currentFrameStartingPoint</a>.
+        1. Set <a>previousVisualRepresentation</a> to the <a>currentVisualRepresentation</a>.
+    1. Call <a>compute the layout shift</a> with <var>unstable elements</var>.
+</div>
+
+Compute the Layout Shift {#sec-compute-layout-shift}
+-----------------------------------------------------
+
+<div algorithm="compute the layout shift">
+    When asked to to <dfn export>compute the layout shift</dfn>, with <var>unstable elements</var>, a list of impacted <a>element</a>'s on the page, as input, run the following steps:
+
+    1. Let <var>impacted region</var> be initially set to <code>null</code>.
+    1. Let <var>max shift fraction</var> be initially set to 0.
+    1. For each tuple <var>tuple</var> in <var>unstable elements</var>:
+        1. Take the geometric union of <var>impacted region</var> with <var>tuple</var>’s <a>currentVisualRepresentation</a> and <var>tuple</var>’s <a>previousVisualRepresentation</a> if <a>previousVisualRepresentation</a> is not null.
+        1. Set <var>max shift fraction</var> equal to <var>shiftFraction</var> if <var>tuple</var>’s <var>shiftFraction</var> > <var>max shift fraction</var>.
+    1. Let <var>impacted fraction</var> be the area of the <var>impacted region</var> divided by the area of the current <a>viewport</a>.
```

Comment:
Once we have linked precisely what rects we're talking about, presumably we'll be able to link to "area".

Reply:
I think the only ambiguity in referring to "area" is that of which coordinate space to use - we can specify the coordinate space of the initial containing block.

Reply:
done.